packages/bitcore-wallet-service/src/lib/server.ts

```diff
@@ -1164,7 +1164,7 @@ export class WalletService implements IWalletService {
           return cb(new ClientError('The wallet you are trying to join was created for a different chain'));
         }
 
-        if (!Utils.compareNetworks(wallet.network, xPubKey.network.name)) {
+        if (!xPubKey.network.is(Utils.getGenericName(wallet.network))) {
```

Comment:
I actually don' think this line needs to change. The Utils.compareNetworks() should handle the translation. As this is, it breaks when wallet.network == regtest